### PR TITLE
Move random key generator to public API

### PR DESCRIPTION
**Pull request by sleeepyjack**
The new key generator is pretty handy when writing standalone PoCs/benchmarks for cuco.
So, I would like to move it to the public API, which can then be used directly from the cuco cmake target (Cc @robertmaynard, who proposed this solution in an offline discussion).

I'm not 100% settled on where exactly to put this file. I'm open to bikeshedding.

This PR sits on top of #274

### CODE REVIEW

File: include/cuco/util/key_generator.hpp [hunk 1]
```diff
@@ -34,21 +34,36 @@
 #include <time.h>
 #include <type_traits>
 
-namespace cuco::benchmark {
+namespace cuco::util {
 
 namespace dist_type {
```
Comment (PointKernel) [inline include/cuco/util/key_generator.hpp]: Actually, is `distribution` better than `dist_type`?

**Review by PointKernel — COMMENTED** [review]
I'm inclined to use `utility` instead of the short term`util` to name the folder and namespace.

File: include/cuco/util/key_generator.hpp (hunk 1 above)
Comment (sleeepyjack) [reply include/cuco/util/key_generator.hpp]: Yep, that's a remnant from the prototype. I'll change it to `distribution`

Comment (sleeepyjack) [reply include/cuco/util/key_generator.hpp]: addressed in [5bfa854](https://github.com/NVIDIA/cuCollections/pull/275/commits/5bfa854e95eb87c07e1dbde0f77c249830678685)

**Comment by sleeepyjack** [comment]
> I'm inclined to use `utility` instead of the short term`util` to name the folder and namespace.

addressed in [3306a95](https://github.com/NVIDIA/cuCollections/pull/275/commits/3306a95f8159732f5f38ce82d4fce5296ef3fce2)

**Review by PointKernel — APPROVED** [review]

**Pull request merged**